erpnext/hr/utils.py

```diff
@@ -482,7 +484,11 @@ def share_doc_with_approver(doc, user):
 		if doc_before_save.get(approver) != doc.get(approver):
 			frappe.share.remove(doc.doctype, doc.name, doc_before_save.get(approver))
 
-def validate_active_employee(employee):
-	if frappe.db.get_value("Employee", employee, "status") == "Inactive":
-		frappe.throw(_("Transactions cannot be created for an Inactive Employee {0}.").format(
+def validate_active_employee(employee: str, on_date: Optional[str] = None):
+	relieving_date, status = frappe.get_value(
+		"Employee", employee, ("relieving_date", "status")
+	)
+
+	if (on_date and relieving_date and relieving_date < getdate(on_date)) or status == "Inactive":
```

Comment:
```suggestion
	if status == "Inactive" and (not relieving_date or (on_date and relieving_date and relieving_date < getdate(on_date)))
```

The condition should be modified to this since the older one will always be true if the status is Inactive irrespective of the relieving date?

In the modified one, we throw the validation if:
- Status is Inactive and Relieving Date is not set. This is necessary for a use case where companies first mark all employees as Inactive and then activate them when their agreement starts or where daily wage workers work on a contract basis and users don't explicitly set the Relieving Date.
- Status is Inactive and the transaction date is after the Relieving Date.

The first condition is covered in unit tests but the second one introduced in this PR isn't. Can we add a test for the same?